libs/utils/target_script.py

```diff
@@ -89,7 +89,11 @@ def push(self):
         :type actions: list(str)
         """
 
-        actions = ['set -e'] + self.commands + ['set +e']
+        self._pid_file = self._target.path.join(self._target.working_directory, 
```

Comment:
Yes, I agree a single kill would be far better. Unfortunately, killing `self._bg_shell` stops the shell _on the host_, but the script keeps running on the target. TBH I'm not sure to really understand why, but I couldn't find any other way of solving the issue ...